common/changes/@itwin/core-react/core-react-checkbox-workaround_2022-02-02-10-47.json

```diff
@@ -0,0 +1,10 @@
+{
```

Comment:
>It is a fix on dev version, how urgent is to deliver it? Cant we just fix it in iTwinUI instead?

The problem exists on all iTwin.js versions starting from 3.0.0-dev.4 up until current 3.1.0-dev.x versions. Stable release of 3.0.0 happened somewhere in between.

This PR prepares us to release the next patch version of iTwin.js 3.0 without additional delays. Upgrading to the next version of @itwin/itwinui-react package will require some coordination because we locked down the package version to 1.18.0 due to some build issue.

Reply:
Would you try something simpler?
https://github.com/iTwin/iTwinUI-react/issues/533#issuecomment-1028332337

Reply:
I would love to used the solution proposed on the bug thread, but it only applies to function components. In a class component the the current workaround is simpler to add and remove.

Reply:
Well, I guess I could localise almost all the code for the proposed workaround into one place, so the other solution isn't too bad also...

Reply:
Well, I guess I could localise almost all the code for the proposed workaround into one place, so the other solution isn't too bad also...

Reply:
I implemented the suggested workaround but it seems that some of our tests that attempt to click the checkbox now fail. I'm not going to edit the existing tests, so the current workaround will remain.